google/cloud/storage/internal/service_account_credentials_test.cc

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/service_account_credentials.h"
+#include "google/cloud/internal/setenv.h"
+#include "google/cloud/storage/internal/credential_constants.h"
+#include "google/cloud/storage/internal/nljson.h"
+#include "google/cloud/storage/internal/time_fetcher.h"
+#include "google/cloud/storage/testing/mock_http_request.h"
+#include <gmock/gmock.h>
+#include <cstring>
+
+namespace google {
+namespace cloud {
+namespace storage {
+namespace testing {
+namespace {
+
+using storage::internal::ServiceAccountCredentials;
+using storage::internal::TimeFetcher;
+using storage::testing::MockHttpRequest;
+using ::testing::_;
+using ::testing::An;
+using ::testing::HasSubstr;
+using ::testing::Invoke;
+using ::testing::Return;
+using ::testing::StrEq;
+
+// This "magic" assertion was generated using logic from an existing crypto
+// library (Python's oauth2client) from the keyfile above. It was slightly
```

Comment:
> from the keyfile above

is that below, as in the `JSON_KEYFILE_CONTENTS`?  And do you mind documenting what commands were used to generate the file with `oauth2client`?  I am guessing it was a short `python -e` script?

Reply:
I actually threw a breakpoint here:
https://github.com/google/oauth2client/blob/0d1c814779c21503307b2f255dabcf24b2a107ac/oauth2client/crypt.py#L99

and substituted in the appropriate values there, where the RSA-SHA256 `signer` had already been initialized for me.  I can probably work up a script to actually make this reproducible.  Left a TODO for this (#771).